### PR TITLE
Filter dropdowns

### PR DESCRIPTION
**Pull request by zachary-kent**
### Summary 

This PR adds basic functionality to the filter bar on the explore courses page; clicking each button now causes a dropdown menu to appear in accordance with the designs created on figma. The checkboxes in the dropdown menus can also be checked and unchecked. I implemented my own basic dictionary and hash set to keep track of which boxes have been checked and retrieve this information in amortized constant time--the number of checkboxes could be large for the dropdown menus such as Majors/Minors. Additionally, I tried to avoid use of "Magic Numbers" through creating type aliases for each of the handlers and component property types I used.

EDIT: I replaced custom Set and Dictionary with the new ES6 Set and Map classes.

### Test Plan 

An example of one of the filter dropdowns is shown below. You can run the code to verify that the other dropdowns behave as expected as well. Note that functionality to show filtered courses has not yet been implmeneted.
image

### CODE REVIEW

File: design-at-cornell/src/explore-courses/title/FilterCheckbox.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { FilterCheckboxProps } from "../types/PropertyTypes";
+import { Checkbox, FilterCheckboxContainer } from "../ExploreCoursesStyles";
+import check from '../../static/images/check.svg'
+
+const FilterCheckbox = ({ label, checked, onClick }: FilterCheckboxProps) => (
```
Comment (katrina433) [inline design-at-cornell/src/explore-courses/title/FilterCheckbox.tsx:6]: Your custom checkboxes work well! HTML also has `<input type="checkbox">` with some built-in functionalities that might decrease some of the functions and types that you need to maintain. But everything is well integrated so I don't think it's necessary to change.

File: design-at-cornell/src/explore-courses/title/FilterDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                {checkboxes}
+            </FilterDropdownScrollContainer>
+            {selectionBubbles}
+            {checkboxData.size > 0 && clearFilters}
+        </FilterDropdownContainer>
+    );
+    return (
+        <FilterDropdownContainer expand={expand}>
+            {checkboxes}
+            {selectionBubbles}
```
Comment (katrina433) [inline design-at-cornell/src/explore-courses/title/FilterDropdown.tsx]: The placement of the selection bubbles is a bit off. Maybe do not include the selection bubbles here where `scroll` is false? When scrolling isn't needed, all labels can be seen at once in the dropdown menu, so I don't think having extra bubbles here is necessary. 

image

**Review by katrina433 — COMMENTED** [review]
Filtering and searching work great! I like how the filter bar displays the number of filters applied. One suggestion is to add styling `cursor: pointer;` for checkboxes and the "Clear All Filters" button to make them more "button-like".

File: design-at-cornell/src/explore-courses/title/FilterCheckbox.tsx (hunk 1 above)
Comment (zachary-kent) [reply design-at-cornell/src/explore-courses/title/FilterCheckbox.tsx:6]: Thanks so much! I tried using the input element at first, but had a lot of difficulty styling them. From what I could find online, styling default checkboxes can be pretty challenging.

File: design-at-cornell/src/explore-courses/title/FilterDropdown.tsx (hunk 2 above)
Comment (zachary-kent) [reply design-at-cornell/src/explore-courses/title/FilterDropdown.tsx]: Thanks, I definitely agree! Just made those changes.

**Review by katrina433 — APPROVED** [review]
Looks good!

**Pull request merged**